src/Workspaces/Remote/ServiceHub/Services/BrokeredServiceBase.cs

```diff
@@ -86,7 +87,7 @@ protected async ValueTask<T> RunServiceAsync<T>(Func<CancellationToken, ValueTas
         protected async ValueTask RunServiceAsync(Func<CancellationToken, ValueTask> implementation, CancellationToken cancellationToken)
         {
             WorkspaceManager.SolutionAssetCache.UpdateLastActivityTime();
-            using var _ = LinkToken(ref cancellationToken);
+            using var combined = cancellationToken.CombineWith(ClientDisconnectedSource.Token);
```

Comment:
⚠️  The code below is expected to pass `combined.Token` to `implementation`, but instead passed `cancellationToken`. I'm going to fix this in a follow-up so I don't reset this build over it.

Reply:
➡️ now fixed